src/docfx/lib/log/Telemetry.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Metrics;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class Telemetry
+    {
+        private static readonly TelemetryClient s_telemetryClient = new TelemetryClient();
+
+        private static readonly string s_version = typeof(Telemetry).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "<null>";
+        private static readonly string s_os = RuntimeInformation.OSDescription ?? "<null>";
+
+        private static string s_repo = "<null>";
+        private static string s_branch = "<null>";
+
+        public static void SetRepository(string repo, string branch)
+        {
+            s_repo = repo ?? "<null>";
+            s_branch = branch ?? "<null>";
+        }
+
+        public static void TrackOperationDuration(string name, TimeSpan duration)
+        {
+            s_telemetryClient
+                .GetMetric(new MetricIdentifier(null, $"time", "name", "os", "version", "repo", "branch"))
+                .TrackValue(duration.TotalMilliseconds, name, s_os, s_version, s_repo, s_branch);
+        }
+
+        public static void TrackErrorCount(string code, ErrorLevel level)
+        {
+            s_telemetryClient
+                .GetMetric(new MetricIdentifier(null, $"error", "code", "level", "os", "version", "repo", "branch"))
+                .TrackValue(1, code, level.ToString(), s_os, s_version, s_repo, s_branch);
+        }
+
+        public static void TrackCacheTotal(CacheName name)
```

Comment:
what's that means?

Reply:
or TrackCacheHit?

Reply:
Tracking Hit/Miss count is hard then tracking Miss/Total count, see GitHubUser cache as an example, there is no clear if/else switch, but it is easier to find all total accesses and missed accesses.